ampersand-react-mixin.js

```diff
@@ -18,14 +18,20 @@ var deferbounce = function (fn) {
 };
 
 module.exports = events.createEmitter({
+
     watch: function (modelOrCollection, opts) {
         var events;
-        if (modelOrCollection.isCollection) {
-            events = 'add remove reset';
-        } else if (modelOrCollection.isState) {
-            events = 'change';
-        } else {
-            return;
+
+        if (typeof modelOrCollection === 'object'){
```

Comment:
I this this might need to be

```js
if (modelOrCollection !== null && typeof modelOrCollection === 'object') {
```

since `typeof null === 'object'` but calling `null.isCollection` will trigger a `TypeError: Cannot read property 'isCollection' of null`.